astropy/wcs/wcsapi/wrappers/compound_wcs.py

```diff
@@ -84,29 +138,56 @@ def world_axis_object_classes(self):
     @property
     def pixel_shape(self):
         if not any(w.array_shape is None for w in self._wcs):
-            return tuplesum(w.pixel_shape for w in self._wcs)
+            pixel_shape = tuplesum(w.pixel_shape for w in self._wcs)
+            out_shape = self.mapping.inverse(*pixel_shape)
+            for i, ix in enumerate(self.mapping.mapping):
+                if out_shape[ix] != pixel_shape[i]:
+                    raise ValueError(
+                        "The pixel shapes of the supplied WCSes do not match for the dimensions shared by the supplied mapping.")
```

Comment:
Shouldn't we validate this in the input to avoid doing this check every time this property is accessed? (same comment for pixel_bounds)

Reply:
fair point, but that's basically doing to duplicate this whole property for what I can't imagine is going to be a big gain?